### ISSUE TITLE
systemd-networkd (v234) fails to configure bonding on boot with (at least) linux 4.12

### ISSUE THREAD

**Issue by q2dg** [systemd/systemd#8774]
### Submission type
 - Bug report

### systemd version the issue has been seen with
> v234 

### Used distribution
> Ubuntu 17.10

### In case of bug report: Expected behaviour you didn't see
If I use following commands to make a two-cable ephemeral bonding in two VMs to connect themselves, everything is ok:

_ip link add elmeubond type bond
ip link set eth0 master elmeubond 
ip link set eth1 master elmeubond
#The other VM has the IP 192.168.1.2
ip address add 192.168.1.1/24 dev elmeubond
ip link set elmeubond up_

But if I create a permanent bonding with the edition of following files, _ip a_ command shows all seems ok (everything is "UP", IP is assigned, etc) but VMs didn't ping themselves!

*File /etc/systemd/network/elmeubond.netdev:
[NetDev]
Name=elmeubond
Kind=bond
[Bond]
Mode=balance-rr

*File: /etc/systemd/network/elmeubond.network:
[Match]
Name=elmeubond
[Network]
Address=192.168.1.2/24
BindCarrier=enp0s8 enp0s9

*File /etc/systemd/network/enp0s8.network:
[Match]
Name=enp0s8
[Network]
Bond=elmeubond

*File /etc/systemd/network/enp0s9.network:
[Match]
Name=enp0s9
[Network]
Bond=elmeubond

I don't know if it's related, but networkctl says individual interfaces are "configuring".  And "bonding" module isn't the problem (I think) because is loaded: it appears in lsmod output.

I've see https://github.com/systemd/systemd/issues/599, which seems similar, is closed but I don't know why.

Thanks a lot

**Comment by q2dg** [comment]
On the other hand (I don't know if it is related to issue explained above), if I remove all *.network and *.netdev files related to the bonding and I restart systemd-networkd service, bonding still is shown when doing a _ip a_ command. It is not until I manually do _rmmod bonding && modprobe bonding_ the (non-functional, anyway) bonding is disappeared.

**Comment by yuwata** [comment]
PR #11681 or #11679 help this?

It may not be related, but why you set `BindCarrier=`?

**Comment by q2dg** [comment]
Thanks!

I don't know if it helps because I don't know how to compile an "ad-hoc" systemd to try these new PR (I haven't enough technical skills). I'll wait until they get merged and released in an new stable version, no problem.

On the other hand, I set BindCarrier just because everyone says it's worth it, but I haven't any other founded reason. See for instance:
https://wiki.archlinux.org/index.php/VLAN
https://clearlinux.org/documentation/clear-linux/guides/network/network-bonding
https://major.io/2015/08/21/using-systemd-networkd-with-bonding-on-rackspaces-onmetal-servers/
https://forum.manjaro.org/t/how-to-create-lacp-802-3ad-bond-using-systemd-networkd/14655

**Comment by q2dg** [comment]
In fact, in https://fedoracloud.readthedocs.io/en/latest/networkd.html#systemd-networkd-use-cases is said this: "The BindCarrier is optional but recommended. It gives systemd-networkd the hint that if both bonded interfaces are offline, it should remove the bond configuration until one of the interfaces comes up again."

**Comment by keszybz** [comment]
It's been ages since this was reported and is hopefully fixed since a long time. Please reopen if this still occurs with recently released versions.
